shared/shot.js

```diff
@@ -0,0 +1,726 @@
+// Note: in this library we can't use any "system" dependencies because this can be used from multiple
+// environments
+/* globals console */
+
+/** Throws an error if the condition isn't true.  Any extra arguments after the condition
+    are used as console.error() arguments. */
+function assert(condition) {
+  if (! condition) {
+    console.error.apply(console, ["Failed assertion:"].concat(Array.prototype.slice.call(arguments, 1)));
```

Comment:
`console.error("Failed assertion", ...args)` along with `...args` as a second argument to `assert`